AK/SyncGenerator.h

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (c) 2025, Dan Klishch <danilklishch@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/Coroutine.h>
+#include <AK/Optional.h>
+#include <AK/ScopeGuard.h>
+
+namespace AK {
+
+template<typename Y>
+class SyncGenerator {
```

Comment:
Another late-breaking question: Does it make sense to mark this `[[nodiscard]]`? Or are there use cases where it's useful to call a function yielding values purely for the side effects? (I can't think of one, but maybe I'm not creative enough.)